### ISSUE TITLE
Add support for grob as image

### ISSUE THREAD

**Issue by jonocarroll** [jonocarroll/ntfy#19]
`get_image_path` currently lacks a logic branch for an input which is not a `ggplot` or a string

https://github.com/jonocarroll/ntfy/blob/125a3bf40d9d482efbe8d5fe19f2300a71f478d5/R/ntfy.R#L17-L28

so a `grob` produces an uninformative error. A default case should be added, but specific support for `grob` would also be good, see also #18.

An example `grob` for testing would be 

```r
library(grid)

rect_grob <- rectGrob(
  x = 0.5, y = 0.5,
  width = 0.3, height = 0.3,
  gp = gpar(fill = "lightblue", col = "navy")
)

circle_grob <- circleGrob(
  x = 0.5, y = 0.5, r = 0.15,
  gp = gpar(fill = "pink", col = "red")
)

text_grob <- textGrob(
  "Hello Grid!",
  x = 0.5, y = 0.8,
  gp = gpar(fontsize = 16, fontface = "bold")
)

# Combine all grobs into a single grob object
my_grob <- gTree(children = gList(rect_grob, circle_grob, text_grob))

grid.newpage()
grid.draw(my_grob)

class(my_grob)
# [1] "gTree" "grob"  "gDesc"
```

![image](https://github.com/user-attachments/assets/9b81ce06-f830-4ee3-b81f-da660c42fc20)

**Comment by hadley** [comment]
IMO the most general type of object to accept here would be what `recordPlot()` returns. That will work with all types of graphics, and make it easy to send the contents of the current plot window.

**Comment by jonocarroll** [comment]
I'm not sure I follow; the API requires linking to a saved file, so the logic here needs to write a file then return the path to it. I agree that getting the contents of the current window would be useful (hence the `ggsave` branch when the current plot is a `ggplot` object), but is there a more general way to write out a `recordedplot` object?

**Comment by hadley** [comment]
You'd do something like this to work with recorded plots:

```R
get_image_path <- function(image) { 
  if (inherits(image, "ggplot")) { 
    requireNamespace("ggplot2", quietly = FALSE) 
    filename <- tempfile(pattern = "gg", fileext = ".png") 
    ggplot2::ggsave(filename, image) 
  } else if (inherits(x, "recordedplot")) {
    filename <- tempfile(pattern = "ntfy-", fileext = ".png") 
    png(filename, width = 800, height = 600)
    replayPlot(image)
    dev.off()
  } else if (is.character(image)) { 
    stopifnot(file.exists(image)) 
    filename <- image 
  } 
   
  return(filename) 
} 
```

**Comment by jonocarroll** [comment]
I've added that logic here https://github.com/jonocarroll/ntfy/blob/5abff69e4fe6d797784b4c065d342f27f4ef8a19/R/ntfy.R#L87-L109

This solution was only part of what this issue needs, though, so I've also added explicit support for a `grob` (stored as a not-yet-plotted object) with similar logic. I also added some documentation about how to trigger the 'use the current plot' logic, which (unless I misunderstand) involves 

```r
ntfy_send(image = recordPlot())
```

I'm not entirely sure how to formally test that because creating a plot within the test generates a `Rplots.pdf`, but testing it interactively, it works with all the options.

**Comment by hadley** [comment]
IMO, it would be better to introduce a new explicit `image()` object that make it easier to provide some of these options, rather than attempting to add to arguments of `ntfy_send()`. I'm happy to sketch something out if you're interested.

**Comment by jonocarroll** [comment]
Yeah, I think given the complexity around it, extracting that makes sense. One consideration would be the namespace clash with `graphics::image`, only because it would raise a masking message. `ntfy_image` might be an alternative. 

Your suggestions for how to structure that would be most welcome.
